Add OPA environment information Gauge to Prometheus metrics

jmoghisi opened the pull request:
Add OPA environment information Gauge to Prometheus metrics

Closes issue https://github.com/open-policy-agent/opa/issues/5852
Provides version of OPA in a Prometheus gauge for observability

### What are the changes in this PR?

A new Prometheus metric has been added that includes a constant label for the version of OPA.

ashutosh-narkar commented on plugins/status/metrics.go:
```diff
@@ -5,6 +5,17 @@ import (
 )
 
 var (
+	createOpaInfoGauge = func(version string) prometheus.Gauge {
```
I think we can get the current OPA version by importing `github.com/open-policy-agent/opa/version` and then using the `Version` field. If that's the case then we can do:

```go
opaInfo := prometheus.NewGauge(
	prometheus.GaugeOpts{
		Name:        "opa_info",
		Help:        "Information about the OPA environment.",
		ConstLabels: map[string]string{"version": version},
	},
    ).Set(1)
```

ashutosh-narkar commented on plugins/status/plugin.go:
```diff
@@ -238,7 +238,10 @@ func (p *Plugin) Start(ctx context.Context) error {
 
 	if p.config.Prometheus && p.manager.PrometheusRegister() != nil {
-		p.register(p.manager.PrometheusRegister(), pluginStatus, loaded, failLoad,
+		version := p.manager.Labels()["version"]
+		opaInfo := createOpaInfoGauge(version)
```
We can avoid this by importing `github.com/open-policy-agent/opa/version` and then using the `Version` field.

ashutosh-narkar commented on plugins/status/plugin_go1.17_test.go:
```diff
@@ -106,6 +111,60 @@ func TestPluginPrometheus(t *testing.T) {
+		t.Fatalf("Expected gauge to have name opa_info but was %s", fqName)
+	}
+
+	versionExp := fixture.manager.Labels()["version"]
```
```go
import  github.com/open-policy-agent/opa/version

versionExp := version.Version
```

ashutosh-narkar reviewed (commented):
@jmoghisi this looks good. Just one comment about the way we get the current OPA version.

ashutosh-narkar commented:
@jmoghisi please let us know if you need any help and when this is ready for review. Thanks!

jmoghisi commented:
> @jmoghisi please let us know if you need any help and when this is ready for review. Thanks!

I addressed the comments, it's ready for review again.

ashutosh-narkar reviewed (approved):
LGTM. Thanks @jmoghisi for the contribution!
